src/mono/mono/metadata/class.c

```diff
@@ -857,21 +857,21 @@ inflate_generic_type (MonoImage *image, MonoType *type, MonoGenericContext *cont
 
 		nt = mono_metadata_type_dup (image, type);
 		nt->type = MONO_TYPE_GENERICINST;
-		nt->data.generic_class = gclass;
+		m_type_data_set_generic_class (nt, gclass);
```

Comment:
Won't mono_metadata_lookup_generic_class return a generic class meaning this could be unchecked? If not then maybe mono_metadata_lookup_generic_class should do that and assert if not returning a generic class.